automation_tools/__init__.py

```diff
@@ -1026,6 +947,129 @@ def product_install(distribution, create_vm=False, certificate_url=None,
         )
 
 
+def satellite5_product_install(create_vm=False, selinux_mode=None):
+    """Task which installs only satellite5 product.
+
+    If ``create_vm`` is True then ``vm_destroy`` and ``vm_create`` tasks will
+    be run. Make sure to set the required environment variables for those
+    tasks. Also, if one of the ``setup_ddns`` required environment variables
+    is set then that task will run.
+
+    :param bool create_vm: creates a virtual machine and then install the
+        product on it. Default: False.
+    :param str selinux_mode: switches to specified selinux mode.
+
+    """
+    # Command-line arguments are passed in as strings.
+    if isinstance(create_vm, str):
+        create_vm = (create_vm.lower() == 'true')
+
+    if selinux_mode is None:
+        selinux_mode = os.environ.get('SELINUX_MODE', 'enforcing')
+
+    if create_vm:
+        target_image = os.environ.get('TARGET_IMAGE')
+        if target_image is None:
+            print('The TARGET_IMAGE environment variable should be defined')
+            sys.exit(1)
+
+        execute(vm_destroy, target_image, delete_image=True)
+        execute(vm_create)
+
+        if 'DDNS_HASH' in os.environ or 'DDNS_PACKAGE_URL' in os.environ:
+            execute(
+                setup_ddns, env['vm_domain'], env['vm_ip'], host=env['vm_ip'])
+
+    # When creating a vm the vm_ip will be set, otherwise use the fabric host
+    host = env.get('vm_ip', env['host'])
+
+    # Register and subscribe machine to Red Hat
+    execute(subscribe, host=host)
+
+    execute(install_prerequisites, host=host)
+    execute(setenforce, selinux_mode, host=host)
+    execute(enable_satellite_repos, host=host)
+    execute(update_packages, warn_only=True)
+    execute(satellite5_installer, host=host)
+
+
+def satellite5_installer():
+    """Installs Satellite 5 from an ISO image.
+
+    The following environment variables affect this command:
+
+    RHN_USERNAME
+        Red Hat Network username.
+    RHN_PASSWORD
+        Red Hat Network password.
+    ISO_URL
+        The URL where the ISO will be downloaded.
+    SATELLITE_CERT_URL
+        The URL where the activation certificate will be downloaded.
+
+    """
+    iso_url = os.environ.get('ISO_URL')
+    if iso_url is None:
+        print('Please provide a valid URL for the ISO image.')
+        sys.exit(1)
+
+    # Download and mount the ISO
+    print('Downloading ISO...')
+    iso_download(iso_url)
+    run('mkdir -p ISO')
+    run('mount -t iso9660 -o loop *.iso ISO')
```

Comment:
In the Satellite 6 ISO job we don't unmount the directory, I don't know the reason but maybe it will be used later. Anyway if we discover that the ISO can be unmounted then we should do the same for Satellite 6 task.